wDeveloper/SetupjDataPortal.md

```diff
@@ -0,0 +1,134 @@
+keywords: [portal, connection, override, command]
+description: The jDataPortal is a powerful Interject function that not only allows users to customize the connection to an Interject Data Portal. In addition to establishing a connection to a Portal, users can also override certain parameters of the Portal to customize their reporting needs.
+---
+* * *
+
+## Overview
+
+The [jDataPortal](/wIndex/jDataPortal.html) is a powerful Interject function that not only allows users to not only establish and customize the connection to an Interject Data Portal, but also override certain parameters of the Portal to customize reporting needs.
+
+### Setup jDataPortal
+
+In this section, you will open a report and change the Data Portal to a jDataPortal.
+
+Begin by opening up the Customer Collections report.
+
+![](/images/SetupjDataPortal/CustomerCollectionsReport.png)
+<br>
+
+In order to access the Interject configuration area, unfreeze the panes via the Quick Tools menu.
+
+![](/images/SetupjDataPortal/UnFreezePanes.png)
+<br>
+
+Select the report function.
+
+![](/images/SetupjDataPortal/ReportFunctionSelected.png)
+<br>
+
+In the formula bar, change the **DataPortal** parameter from "Northwind Customers" to:
+
+```
+jDataPortal("Northwind Customers")
+```
+
+![](/images/SetupjDataPortal/FormulaBar.png)
+<br>
+
+Enter "Market" for the Company Name filter and Pull the data.
+
+![](/images/SetupjDataPortal/PullData.png)
+<br>
+
+### Result Sets
+
+The **DataResultNumber** is the second parameter for the jDataPortal. This parameter specifies the particular data set you want returned. For instance, if the stored procedure for the data portal returns 4 queries, entering 2 for this parameter will return the 2nd query to your report.
+
+You can also enter negative numbers to specify the query in relation to the last query. For example, -1 will return the last query listed in your stored procedure. To illustrate, the following chart shows an example for a stored procedure returning 4 data sets:
+
+| Query<br>Position | Postive<br>Reference | Negative<br>Reference |
+|---|---|---|
+| Query 1 | 1 | -4 |
+| Query 2 | 2 | -3 |
+| Query 3 | 3 | -2 |
+| Query 4 | 4 | -1 |
+
+### Order By
+
+The **OrderBy** parameter accepts comma separated values of column names that will be used to sort the data result set. In this section you will add a value for this parameter to order the results.
+
+First select the jDataPortal function by putting the cursor inside the name and click the **fx** button to bring up Excel's Function Wizard.
+
+![](/images/SetupjDataPortal/SelectFx.png)
+<br>
+
+In the **OrderBy** field, enter:
+
+```
+"[CustomerName] DESC"
+```
+
+![](/images/SetupjDataPortal/AddOrderBy.png)
+<br>
+
+Now Pull the data again and notice the results are in descending order by Company Name.
+
+![](/images/SetupjDataPortal/ResultsOrdered.png)
+<br>
+
+The **OrderBy** parameter excepts multiple column names (e.g. "CompanyName, ContactTitle). Some key points:
```

Comment:
should this be "accepts"? And should the quotations at CompanyName be closed?

Reply:
Yes I will fix this thanks!